impl/src/main/java/org/jboss/cdi/tck/tests/extensions/beanDiscovery/event/ordering/ProductManagement.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.cdi.tck.tests.extensions.beanDiscovery.event.ordering;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessBeanAttributes;
+import javax.enterprise.inject.spi.ProcessInjectionPoint;
+import javax.enterprise.inject.spi.ProcessInjectionTarget;
+import javax.enterprise.inject.spi.ProcessManagedBean;
+import javax.enterprise.inject.spi.ProcessProducer;
+import javax.enterprise.inject.spi.ProcessProducerMethod;
+
+/**
+ * CDI defines this order (see Bean Discovery chapter) for producers: PIP -> PP -> PBA -> PPM (e.g. ProcessBean for producer
+ * methods) And following order for managed beans: PIP -> PIT -> PBA -> PB
+ *
+ * Of course this can get a lot more complex, but let's cover this basic case.
+ *
+ * @author <a href="mailto:manovotn@redhat.com">Matej Novotny</a>
+ */
+public class ProductManagement implements Extension {
+
+    private List<Object> listOfProducerEvents = new ArrayList<>();
+    private List<Object> listOfBeanEvents = new ArrayList<>();
+
+    // producer method observers
+    public void observePIP(@Observes ProcessInjectionPoint<PoorWorker, String> pip) {
+        listOfProducerEvents.add(pip);
```

Comment:
Maybe we could take use of [ActionSequence](https://github.com/cdi-spec/cdi-tck/blob/master/impl/src/main/java/org/jboss/cdi/tck/util/ActionSequence.java) and store only class information to avoid this `instanceof` check.